routers/api/packages/pypi/pypi.go

```diff
@@ -25,8 +25,8 @@ import (
 var normalizer = strings.NewReplacer(".", "-", "_", "-")
 var nameMatcher = regexp.MustCompile(`\A[a-zA-Z0-9\.\-_]+\z`)
 
-// https://www.python.org/dev/peps/pep-0440/#appendix-b-parsing-version-strings-with-regular-expressions
-var versionMatcher = regexp.MustCompile(`^([1-9][0-9]*!)?(0|[1-9][0-9]*)(\.(0|[1-9][0-9]*))*((a|b|rc)(0|[1-9][0-9]*))?(\.post(0|[1-9][0-9]*))?(\.dev(0|[1-9][0-9]*))?$`)
+// https://github.com/pypa/packaging/blob/0031046f7fad649580bc3127d1cef9157da0dd79/packaging/version.py#L113-L142
+var versionMatcher = regexp.MustCompile(`\Av?(?:(?:(?:[0-9]+)!)?(?:[0-9]+(?:\.[0-9]+)*)(?:[-_\.]?(?:a|b|c|rc|alpha|beta|pre|preview)[-_\.]?[0-9]*)?(?:(?:-[0-9]+)|(?:[-_\.]?(?:post|rev|r)[-_\.]?[0-9]*))?(?:[-_\.]?dev[-_\.]?[0-9]*)?)(?:\+(?:[a-z0-9]+(?:[-_\.][a-z0-9]+)*))?\z`)
```

Comment:
I have absolutely no idea if this regex is correct, but I trust you in this case.
This is one of the regexes you use when you don't want someone to know what you're doing…

Reply:
But perhaps, a comment would be nice to detail what exactly it all matches in general.

Reply:
I did not create the regex. The regex is from the official Python docs. Have a look at the link in the comment above it.